src/Event.php

```diff
@@ -110,6 +134,7 @@ public function getParams()
      * If the parameter does not exist, the $default value will be returned.
      *
      * @param  string|int $name
+     * @psalm-param key-of<TParams> $name
```

Comment:
`TParams` is `array|ArrayAccess|object` here, so I don't think we can really enforce `key-of` on it (won't work on `ArrayAccess` in first place)